packages/nova-base-components/lib/common/Newsletter.jsx

```diff
@@ -1,70 +1,63 @@
 import React, { PropTypes, Component } from 'react';
 import Formsy from 'formsy-react';
-import FRC from 'formsy-react-components';
+import { Input } from 'formsy-react-components';
 import Actions from "../actions.js";
 import { Button } from 'react-bootstrap';
-import Cookie from 'js-cookie';
+import Cookie from 'react-cookie';
 
 import { Messages } from "meteor/nova:core";
 
-const Input = FRC.Input;
-
 
 class Newsletter extends Component {
 
   constructor(props, context) {
     super(props);
     this.subscribeEmail = this.subscribeEmail.bind(this);
-    this.subscribeUser = this.subscribeUser.bind(this);
+    this.successCallbackSubscription = this.successCallbackSubscription.bind(this);
     this.dismissBanner = this.dismissBanner.bind(this);
 
-    const showBanner = 
-      !(Meteor.isClient && Cookie.get('showBanner') === "no") &&
-      Users.getSetting(context.currentUser, 'newsletter_showBanner', true) &&
+    const showBanner =
+      Cookie.load('showBanner') !== "no" &&
       !Users.getSetting(context.currentUser, 'newsletter_subscribeToNewsletter', false);
 
     this.state = {
       showBanner: showBanner
     };
   }
 
+  componentWillReceiveProps(nextProps, nextContext) {
```

Comment:
update the banner reactively! my first "context question" on the slackroom was about that... 💃  but in fact my problem was just related to handling the component lifecycle, not about the context!!! 😄